lib/services/namespace_service.ts

```diff
@@ -0,0 +1,131 @@
+import { PureService } from '@Services/pure_service';
+import { DeviceInterface } from '@Lib/device_interface';
+import { Uuid } from '@Lib/uuid';
+import { UuidString } from './../types';
+
+export type SNNamespace = {
+  identifier: string | UuidString,
+  userUuid?: UuidString,
+  label: string,
+  isDefault: boolean,
+};
+
+const NAMESPACES_STORAGE_KEY = 'namespaces';
+
+/**
+ * The namespace service is responsible of setting the namespace to be used by
+ * the DeviceInterface.
+ */
+export class SNNamespaceService extends PureService {
+  private namespace?: SNNamespace
+
+  constructor(deviceInterface: DeviceInterface, namespaceIdentifier?: string) {
+    super();
+    this.deviceInterface = deviceInterface;
+
+    if (namespaceIdentifier) {
+      const namespace: SNNamespace = {
+        identifier: namespaceIdentifier,
+        label: namespaceIdentifier,
+        isDefault: true
+      };
+      this.namespace = namespace;
+      this.deviceInterface!.switchToNamespace(namespace);
```

Comment:
This function will indirectly write to storage. We shouldn't be writing to storage in a constructor, since assumedly, nothing has changed. Is this assessment correct? If so, you need to separate what's inside `switchToNamespace` to extract the non-async part. You should never call an async function in a constructor, even if you don't need to await the result.

Reply:
Oh wait a minute. This is using deviceInterface.switchToNamespace. I thought it was using this class's function. In that case let me re-analyze.